comment.js

```diff
@@ -1,34 +1,80 @@
 import {getCommentPrefix} from './identifier';
 
-export async function findMatchingComment({octokit, owner, repo, issue_number, identifier}) {
-  const matchingComments = await findMatchingComments({octokit, owner, repo, issue_number, identifier});
+const issueCommenter = {
+    createComment: async function(octokit, context, data) {
+        data.issue_number = context;
+        await octokit.issues.createComment(data);
+    },
+    deleteComment: async function(octokit, data) {
+        data.issue_number = context;
+        await octokit.issues.deleteComment(data);
+    },
+    updateComment: async function(octokit, data) {
+        data.issue_number = context;
+        await octokit.issues.updateComment(data);
+    },
+    listComments: async function(octokit, context, data) {
+        data.issue_number = context;
+        await octokit.issues.listComments(data);
+    }
+}
+
+const commitCommenter = {
+    createComment: async function(octokit, context, data) {
+        data.commit_sha = context;
+        await octokit.repos.createCommitComment(data);
+    },
+    deleteComment: async function(octokit, data) {
+        data.commit_sha = context;
+        await octokit.repos.deleteCommitComment(data);
+    },
+    updateComment: async function(octokit, data) {
+        data.commit_sha = context;
+        await octokit.repos.updateCommitComment(data);
+    },
+    listComments: async function(octokit, context, data) {
+        data.commit_sha = context;
+        await octokit.repos.listCommitComments(data);
+    }
+}
+
+export async getCommentMethod(number, commitSHA) {
+    if ((number && commitSHA) || (!number && !commitSHA)) {
+        throw "Invalid"
+    }
+    if(!number) {
+        return commitCommenter;
+    } else {
+        return issueCommenter;
+    }
+}
```

Comment:
Here we will be giving precedence to issue/pr number in case both of the fields are present. Do we want that behaviour?

Reply:
Nope. Let's throw an error when both of the fields are present. (ignore my earlier comment)

But let's alter the condition to have `if (number)` instead of `if (!number)`.